lib/gears/timer.lua

```diff
@@ -150,17 +150,24 @@ end
 local delayed_calls = {}
 capi.awesome.connect_signal("refresh", function()
     for _, callback in ipairs(delayed_calls) do
-        protected_call(unpack(callback))
+        if #callback > 0 then
+            protected_call(unpack(callback))
+        end
     end
     delayed_calls = {}
 end)
 
 --- Call the given function at the end of the current main loop iteration
 -- @tparam function callback The function that should be called
 -- @param ... Arguments to the callback function
+-- @treturn table All callbacks. This table can be modified during the current
+--  event loop. If all callbacks are removed, then it is cancelled.
```

Comment:
Uhm... this returns a table with entry 1 being the functions to call and the other (array) entries are its arguments. That's (a) a bad thing to return (as an API) and (b) not really what the docs describe.

If you want something unique to return, I'd do something like this:
```lua
function timer.delayed_call(callback, ...)
  local args = { callback, ..., _internal_unique_id = { _internal_cancelled = false } }
  table.insert(delayed_calls, args)
  return args._internal_unique_id
end

function timer.cancel_delayed_call(identifier)
  identifier._internal_cancelled = true
end

capi.awesome.connect_signal("refresh", function()
  for _, callback in ipairs(delayed_calls) do
    if not callback._internal_unique_id._internal_cancelled then
      protected_call(unpack(callback))
    end
  end
  delayed_calls = {}
end)
```